devicetypes/smartthings/aeon-led-bulb-6.src/aeon-led-bulb-6.groovy

```diff
@@ -0,0 +1,337 @@
+		def result = zwaveEvent(encapsulatedCommand)
+		result = result.collect {
+			if (it instanceof physicalgraph.device.HubAction && !it.toString().startsWith("9881")) {
+				response(cmd.CMD + "00" + it.toString())
+			} else {
+				it
+			}
+		}
+		result
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	log.debug "got ConfigurationReport: $cmd"
+	[createEvent(name: "colorTemperature", value: cmd.scaledConfigurationValue)]
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def linkText = device.label ?: device.name
+	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
+}
+
+def buildOffOnEvent(cmd){
+	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
+}
+
+def on() {
+	commands(buildOffOnEvent(0xFF), 5000)
+}
+
+def off() {
+	commands(buildOffOnEvent(0x00), 5000)
+}
+
+def refresh() {
+	commands([zwave.switchMultilevelV3.switchMultilevelGet()] + queryAllColors())
+}
+
+def ping() {
+	log.debug "ping().."
+	unschedule(offlinePing)
+	runEvery5Minutes(offlinePing)
+	command(zwave.switchMultilevelV3.switchMultilevelGet())
+}
+
+def offlinePing() {
+	log.debug "offlinePing()..."
+	sendHubCommand(new physicalgraph.device.HubAction(command(zwave.switchMultilevelV3.switchMultilevelGet())))
+}
+
+def setLevel(level) {
+	setLevel(level, 1)
+}
+
+def setLevel(level, duration) {
+	log.debug "setLevel($level, $duration)"
+	if(level > 99) level = 99
+	commands([
+		zwave.switchMultilevelV3.switchMultilevelSet(value: level, dimmingDuration: duration),
+		zwave.switchMultilevelV3.switchMultilevelGet(),
+	], (duration && duration < 12) ? (duration * 1000) : 3500)
+}
+
+def setSaturation(percent) {
+	log.debug "setSaturation($percent)"
+	setColor(saturation: percent)
+}
+
+def setHue(value) {
+	log.debug "setHue($value)"
+	setColor(hue: value)
+}
+
+def setColor(value) {
+	log.debug "setColor($value)"
+	def result = []
+	if (value.hex) {
+		def c = value.hex.findAll(/[0-9a-fA-F]{2}/).collect { Integer.parseInt(it, 16) }
+		result << zwave.switchColorV3.switchColorSet(red:c[0], green:c[1], blue:c[2], warmWhite:0, coldWhite:0)
+	} else {
+		def hue = value.hue ?: device.currentValue("hue")
+		def saturation = value.saturation ?: device.currentValue("saturation")
+		if(hue == null) hue = 13
+		if(saturation == null) saturation = 13
+		def rgb = huesatToRGB(hue, saturation)
+		result << zwave.switchColorV3.switchColorSet(red: rgb[0], green: rgb[1], blue: rgb[2], warmWhite:0, coldWhite:0)
+	}
+	commands(result) + "delay 7000" + commands(queryAllColors(), 1000)
+}
+
+private getCOLOR_TEMP_MAX() { 6500 }
+private getCOLOR_TEMP_MIN() { 2700 }
+private getCOLOR_TEMP_DIFF() { COLOR_TEMP_MAX - COLOR_TEMP_MIN }
+private getWARM_WHITE_CONFIG() { 0x51 }
+private getCOLD_WHITE_CONFIG() { 0x52 }
+
+def setColorTemperature(temp) {
+	if(temp > COLOR_TEMP_MAX)
+		temp = COLOR_TEMP_MAX
+	else if(temp < COLOR_TEMP_MIN)
+		temp = COLOR_TEMP_MIN
```

Comment:
based on how you're using `temp` it doesn't seem like you need to clamp it

Reply:
Yeah, this is probably overkill since we control the calling code.